src/main/java/org/reaktivity/nukleus/http/internal/routable/stream/TargetOutputEstablishedStreamFactory.java

```diff
@@ -239,14 +235,14 @@ private void processBegin(
                     headers = headers0;
                 }
 
-                this.sourceId = newSourceId;
-                this.target = newTarget;
-                this.targetId = newTargetId;
+                if (!targetStream.started)
+                {
+                    target.doBegin(targetStream.streamId, 0L, correlation.id());
+                    targetStream.started = true;
+                }
+                target.setThrottle(targetStream.streamId, this::handleThrottle);
```

Comment:
Later on from line 275 the throttle handler is either unset ( response headers too long case) or set to throttleBeforeHeadersWritten, in which case method useTargetWindowToWriteResponseHeaders is always called to write out the response headers. That method changes the throttleState to throttleNextWindow when the response has completed, which just propagates target window to source. source should write and END to us at the end of the response, in which case everything gets cleaned up.